NextcloudTalk/BaseChatViewController.swift

```diff
@@ -1590,52 +1634,83 @@ import QuickLook
         audioFileName += ".mp3"
 
         let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+        NCAPIController.sharedInstance().setupNCCommunication(for: activeAccount)
+        let chatFileController = NCChatFileController()
+        chatFileController.initDownloadDirectory(for: activeAccount)
+
+        let tempDirectoryURL = URL(fileURLWithPath: chatFileController.tempDirectoryPath)
+        let destinationFilePath = tempDirectoryURL.appendingPathComponent(audioFileName).path
+
+        let temporaryMessage = self.createTemporaryMessage(
+            message: audioFileName,
+            replyTo: nil,
+            messageParameters: "\(destinationFilePath)",
+            silently: false,
+            isVoiceMessage: true
+        )
+
+        chatFileController.setFileStatusFrom(temporaryMessage.file())
+
+        let movedFileToTemporaryDirectory = chatFileController.moveFileToTemporaryDirectory(
+            fromSourcePath: self.recorder!.url.path,
+            destinationPath: destinationFilePath
+        )
+
+        if movedFileToTemporaryDirectory, NCDatabaseManager.sharedInstance().roomHasTalkCapability(kCapabilityChatReferenceId, for: room) {
+            self.appendTemporaryMessage(temporaryMessage: temporaryMessage)
+        }
+
         NCAPIController.sharedInstance().uniqueNameForFileUpload(withName: audioFileName, originalName: true, for: activeAccount, withCompletionBlock: { fileServerURL, fileServerPath, _, _ in
             if let fileServerURL, let fileServerPath, let recorder = self.recorder {
                 let talkMetaData: [String: String] = ["messageType": "voice-message"]
-                self.uploadFileAtPath(localPath: recorder.url.path, withFileServerURL: fileServerURL, andFileServerPath: fileServerPath, withMetaData: talkMetaData)
+
+                self.uploadFileAtPath(localPath: recorder.url.path, withFileServerURL: fileServerURL, andFileServerPath: fileServerPath, withMetaData: talkMetaData, temporaryMessage: temporaryMessage)
```

Comment:
```suggestion
                self.uploadFileAtPath(localPath: destinationFilePath, withFileServerURL: fileServerURL, andFileServerPath: fileServerPath, withMetaData: talkMetaData, temporaryMessage: temporaryMessage)
```

Otherwise the file is not found on upload (because we moved it before) and we get a 0 byte file.